chain-impl-mockchain/tests/common/address.rs

```diff
@@ -1,14 +1,30 @@
 use chain_addr::{Address, Discrimination, Kind, KindType};
-use chain_impl_mockchain::account::SpendingCounter;
-use chain_impl_mockchain::key::{EitherEd25519SecretKey, SpendingPublicKey};
+use chain_impl_mockchain::{
+    account::SpendingCounter,
+    key::{EitherEd25519SecretKey, SpendingPublicKey},
+    transaction::{Input, Output, UtxoPointer},
+    value::Value,
+};
+use std::fmt::{Debug, Formatter, Result};
 
+#[derive(Clone)]
 pub struct AddressData {
     pub private_key: EitherEd25519SecretKey,
     pub public_key: SpendingPublicKey,
     pub spending_counter: Option<SpendingCounter>,
     pub address: Address,
 }
 
+impl Debug for AddressData {
+    fn fmt(&self, f: &mut Formatter) -> Result {
+        write!(
+            f,
+            "[{:?},{:?},{:?}]",
+            self.public_key, self.spending_counter, self.address
```

Comment:
Typically structs are written out using the `debug_struct` helper method on `Formatter` so that the fields are named in the output.